### PR TITLE
Update react-props-in-constructor-state-is-an-anti-pattern.md

### PR DESCRIPTION
**Pull request by nem035**
Cleanup the text to make the anti-pattern behavior clearer, add some examples, update both questions.

The revision question can become a discussion insight.

Idea:

```md
Is using `props` to setup the `state` in the constructor a useful pattern in React? Can you think of example when it's not? 

???

- it's an anti-pattern
- it's a good practice
- can cause application crashes
- it's useful when working with many props
```

### CODE REVIEW

**Comment by enki-hq** [comment]
### Changelog

| File Name | Error Message |
|:-:|:-:|
| Pull Request | Couldn't find a CHANGELOG entry for this pull request. |

> If you notice anything wrong with these rules contact andrei@enki.com.

File: react/react-intermediary/props-ii/react-props-in-constructor-state-is-an-anti-pattern.md [hunk 1]
```diff
@@ -19,43 +19,63 @@ revisionQuestion:
...
-        {this.state.nameWithQualifier}
+        {this.state.countryName}
       </div>
     );
   }
 }
 ```
 
-This is bad due to `this.state` being invoked when the component is first created, therefore the example allows a value to get out of sync.
+If the parent component of `CountryName` changes the value for `countryCode` and sends it to `CountryName`, `CountryName` will not render it.
```
Comment (nem035) [inline react/react-intermediary/props-ii/react-props-in-constructor-state-is-an-anti-pattern.md:55]:
```suggestion
If the parent component of `CountryName` changes the value for `countryCode` and sends it to `CountryName`, `CountryName` will still show the previous value and not render the update.
```

**Review by kapnobatai137 — DISMISSED** [review]

**Comment by enki-hq** [comment]
### Changelog

| File Name | Error Message |
|:-:|:-:|
| Pull Request | Couldn't find a CHANGELOG entry for this pull request. |

> If you notice anything wrong with these rules contact andrei@enki.com.

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/react/react-intermediary/props-ii/react-props-in-constructor-state-is-an-anti-pattern.md](https://app.enkipro.com/public/preview-insight/enkidevs/curriculum/react/react-intermediary/props-ii/react-props-in-constructor-state-is-an-anti-pattern.md?ref=b16773a557ebd5ae4beb9773e8fafdda9e30fd35)

**Review by kapnobatai137 — APPROVED** [review]

**Review by kapnobatai137 — APPROVED** [review]

**Pull request merged**